jwtclient/transport.go

```diff
@@ -0,0 +1,217 @@
+package jwtclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// JWTTransport is a custom http.Client Transport which implements RoundTripper and adds JWT to headers.
+type JWTTransport struct {
```

Comment:
ok i am not exporting it. But could you elaborate why is it evil?
if we do the type assertion like this: `jwt, ok := client.Transport.(JWTTransport)` and then check for `ok` to be true, we are not risking to cause panic.